source/tutorials/nixos-configuration-on-vm.md

```diff
@@ -0,0 +1,177 @@
+
+[^nixosconf]: This [configuration template](https://github.com/NixOS/nixpkgs/blob/b4093a24a868708c06d93e9edf13de0b3228b9c7/nixos/modules/installer/tools/tools.nix#L122-L226) is used.
+
+```shell-session
+nixos-generate-config --dir ./
+```
+
+In the working directory you will then find two files: `configuration.nix` and `hardware-configuration.nix`.
+
+`hardware-configuration.nix` is specific to the hardware `nix-generate-config` is being run on.
+We can ignore that file for this tutorial because it has no effect inside a virtual machine.
+
+The `configuration.nix` file contains various suggestions for the initial setup of a desktop computer.
+Without comments the configuration file contains the following content:
+
+```nix
+{ config, pkgs, ... }:
+{
+  imports =  [ ./hardware-configuration.nix ];
+
+  boot.loader.systemd-boot.enable = true;
+  boot.loader.efi.canTouchEfiVariables = true;
+
+  services.xserver.enable = true;
+
+  services.xserver.displayManager.gdm.enable = true;
+  services.xserver.desktopManager.gnome.enable = true;
+
+  system.stateVersion = "22.05";
+}
+```
+
+In this tutorial we focus on testing NixOS configurations on a virtual machine.
+Therefore we will remove the reference to `hardware-configuration.nix`:
+
+```diff
+-  imports =  [ ./hardware-configuration.nix ];
+```
+
+Changes to the configuration need to be positioned inside the curly bracket.[^bracket]
+
+[^bracket]: As a reminder `configuration.nix` contains a function that returns an [attribute set](https://nixos.org/manual/nix/stable/language/values.html#attribute-set) that follows the convention of a [module](https://nixos.org/manual/nixos/stable/index.html#sec-writing-modules).
+In the attribute set you describe how you want your NixOS system configured.
+
+To be able to log in add the following lines:[^user]
+
+[^user]: On NixOS your configuration generated using `nix-generate-config` contains the user configuration commented out.
+
+```nix
+  users.users.alice = {
+    isNormalUser = true;
+    extraGroups = [ "wheel" ];
+    packages = with pkgs; [
+      firefox
+      thunderbird
+    ];
+  };
+```
+
+Additionally, you need to specify a password for this user.
+For the purpose of demonstration only, we use specify an insecure, plain text password by adding the `initialPassword` option to the user configuration:[^password]
+
+[^password]: Warning: Do not use plain text passwords outside of this example unless you know what you are doing. See [`initialHashedPassword`] or [`ssh.authorizedKeys`] for more secure alternatives.
+
+```nix
+initialPassword = "testpw";
+```
+
+The complete `configuration.nix` file now looks like this:
+
+```nix
+{ config, pkgs, ... }:
+{
+  boot.loader.systemd-boot.enable = true;
+  boot.loader.efi.canTouchEfiVariables = true;
+
+  services.xserver.enable = true;
+
+  services.xserver.displayManager.gdm.enable = true;
+  services.xserver.desktopManager.gnome.enable = true;
```

Comment:
Just an idea, so this is optional: To get a much smaller image one may want to remove the GUI (and Firefox/Thunderbird as well). Since we're already expecting terminal skills, it may be just as good to present a barebones machine with some random terminal program to show that Nix actually makes it available.